drivers/serial/uart_kb1200.c

```diff
@@ -0,0 +1,684 @@
+{
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	kb1200_uart_fifo_fill(dev, &c, 1);
+#else
+	const struct kb1200_uart_config *config = dev->config;
+	SER_T *ser = (SER_T *)config->base_addr;
+	/* Wait	Tx FIFO	not Full*/
+	while (ser->SERSTS & SER_TxFull)
+		;
+	/* Put a character into	Tx FIFO	*/
+	ser->SERTBUF = c;
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN	*/
+}
+/**********************************************************************************
+ * Read	a 16-bit dat from the device for input.
+ * Function: int uart_poll_in_u16 (const struct	device *dev, uint16_t *p_u16)
+ *
+ * Write a 16-bit datum	to the device for output.
+ * Function: void uart_poll_out_u16 (const struct device *dev, uint16_t	out_u16)
+ **********************************************************************************/
+/*(ene kb1200 function not support)*/
+
+static const struct uart_driver_api kb1200_uart_api = {
+	.poll_in = kb1200_uart_poll_in,
+	.poll_out = kb1200_uart_poll_out,
+	.err_check = kb1200_uart_err_check,
+#ifdef CONFIG_UART_USE_RUNTIME_CONFIGURE
+	.configure = kb1200_uart_configure,
+	.config_get = kb1200_uart_config_get,
+#endif /* CONFIG_UART_USE_RUNTIME_CONFIGURE */
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	.fifo_fill = kb1200_uart_fifo_fill,
+	.fifo_read = kb1200_uart_fifo_read,
+	.irq_tx_enable = kb1200_uart_irq_tx_enable,
+	.irq_tx_disable = kb1200_uart_irq_tx_disable,
+	.irq_tx_ready = kb1200_uart_irq_tx_ready,
+	.irq_rx_enable = kb1200_uart_irq_rx_enable,
+	.irq_rx_disable = kb1200_uart_irq_rx_disable,
+	.irq_rx_ready = kb1200_uart_irq_rx_ready,
+	.irq_err_enable = kb1200_uart_irq_err_enable,
+	.irq_err_disable = kb1200_uart_irq_err_disable,
+	.irq_is_pending = kb1200_uart_irq_is_pending,
+	.irq_update = kb1200_uart_irq_update,
+	.irq_callback_set = kb1200_uart_irq_callback_set
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+};
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+
+/* GPIO module instances */
+#define KB1200_UART_DEV(inst) DEVICE_DT_INST_GET(inst),
+static const struct device *const uart_devices[] = {DT_INST_FOREACH_STATUS_OKAY(KB1200_UART_DEV)};
+static void kb1200_uart_isr_wrap(const struct device *dev)
+{
+	for (int i = 0; i < ARRAY_SIZE(uart_devices); i++) {
+		const struct device *dev_ = uart_devices[i];
+		const struct kb1200_uart_config *config = dev_->config;
+		SER_T *ser = (SER_T *)config->base_addr;
+		if (ser->SERIE & ser->SERPF) {
+			kb1200_uart_irq_handler(dev_);
+		}
+	}
+}
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN*/
+
+static int kb1200_uart_init(const struct device *dev)
+{
+	int ret = 0;
+	const struct kb1200_uart_config *config = dev->config;
+
+	ret = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_DEFAULT);
+	if (ret != 0) {
+		return ret;
+	}
+
+	struct uart_config kb1200_uart_cfg = {.baudrate = 115200,
+					      .parity = UART_CFG_PARITY_NONE,
+					      .stop_bits = UART_CFG_STOP_BITS_1,
+					      .data_bits = UART_CFG_DATA_BITS_8,
+					      .flow_ctrl = UART_CFG_FLOW_CTRL_NONE};
```

Comment:
having a comma at the end of these would help clang-format to work properly:
```suggestion
	struct uart_config kb1200_uart_cfg = {
		.baudrate = 115200,
		.parity = UART_CFG_PARITY_NONE,
		.stop_bits = UART_CFG_STOP_BITS_1,
		.data_bits = UART_CFG_DATA_BITS_8,
		.flow_ctrl = UART_CFG_FLOW_CTRL_NONE,
	};
```